airflow/README.md

```diff
@@ -0,0 +1,182 @@
+# Agent Check: Airflow
+
+## Overview
+
+The Datadog Agent collects many metrics from Airflow, including those for:
+
+- DAGs (Directed Acyclic Graphs): Number of DAG processes, DAG bag size, etc.
+- Tasks: Task failures, successes, killed, etc.
+- Pools: Open slots, used slots, etc.
+- Executors: Open slots, queued tasks, running tasks, etc.
+
+Metrics are collected through the [Airflow StatsD](https://airflow.apache.org/docs/stable/metrics.html) plugin and sent to Datadog's [DogStatsD][8].
+
+In addition to metrics, the Datadog Agent also sends service checks related to Airflow's health.
+
+## Setup
+
+### Installation
+
+All three steps below are needed for the Airflow integration to work properly. Before you begin, you [install the Datadog Agent][9] version `>=6.17` or `>=7.17` that includes Statsd/DogStatsD Mapping feature. 
+
+#### Step 1: Configure Airflow to collect health metrics and service checks
+
+Configure the Airflow check included in the [Datadog Agent][2] package to collect health metrics and service checks.
+
+Edit the `airflow.d/conf.yaml` file, in the `conf.d/` folder at the root of your Agent's configuration directory to start collecting your Airflow service checks. See the [sample airflow.d/conf.yaml][3] for all available configuration options.
+
+#### Step 2: Connect Airflow to DogStatsD (included in the Datadog Agent) by using Airflow `statsd` feature to collect metrics
+
+1. Install the [Airflow StatsD plugin][1].
+
+    ```
+    pip install 'apache-airflow[statsd]'
+    ```
+
+2. Update the Airflow configuration file `airflow.cfg` by adding the following configs:
+
+    ```
+    [scheduler]
+    statsd_on = True
+    statsd_host = localhost
+    statsd_port = 8125
+    statsd_prefix = airflow
+    ```
+
+3. Update the [Datadog Agent main configuration file][10] `datadog.yaml` by adding the following configs:
+
+    ```yaml
+    # dogstatsd_mapper_cache_size: 1000  # default to 1000
+    dogstatsd_mapper_profiles:
+      - name: airflow
+        prefix: 'airflow.'
+        mappings:
+          - match: 'airflow.*_start'
+            name: 'airflow.job.start'
+            tags:
+              job_name: '$1'
+          - match: 'airflow.*_end'
+            name: 'airflow.job.end'
+            tags:
+              job_name: '$1'
+          - match: 'airflow.operator_failures_*'
+            name: 'airflow.operator_failures'
+            tags:
+              operator_name: '$1'
+          - match: 'airflow.operator_successes_*'
+            name: 'airflow.operator_successes'
+            tags:
+              operator_name: '$1'
+          - match: 'airflow\.dag_processing\.last_runtime\.(.*)'
```

Comment:
Why do we need a regex here (and elsewhere)? Asking as I see we use `airflow.pool.open_slots.*` below, so `airflow.dag_processing.last_runtime.*` should be supported?

Reply:
That's because `dag_file` represent a path that can contain `.` character. Hence using `airflow.dag_processing.last_runtime.*` won't work.

All metrics with `dag_file` are using regex.

Reply:
Gotcha, thanks!